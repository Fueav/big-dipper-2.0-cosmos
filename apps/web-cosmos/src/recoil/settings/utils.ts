import { themeDictionary } from '@styles';
import chainConfig from 'ui/chainConfig';
import { Theme } from './types';

// ================================
// CONSTANTS
// ================================

export const THEME_LIST: Theme[] = chainConfig.themes.themeList;

export const THEME_DICTIONARY = themeDictionary;

export const getThemeTemplate = (theme: Theme) => {
  if (THEME_DICTIONARY[theme]) {
    return THEME_DICTIONARY[theme];
  }
  const first = Object.keys(THEME_DICTIONARY)[0];
  if (!first) throw new Error('No theme found');
  return first;
};

export const DATE_LIST = ['locale', 'utc'];

export const TX_LIST = ['compact', 'detailed'];
