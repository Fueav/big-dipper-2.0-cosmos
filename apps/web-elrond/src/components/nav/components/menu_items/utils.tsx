import React from 'react';
import BlockIcon from 'shared-utils/assets/icon-block.svg';
import HomeIcon from 'shared-utils/assets/icon-home.svg';
import TransactionIcon from 'shared-utils/assets/icon-transaction.svg';
import TokenIcon from 'shared-utils/assets/icon-token.svg';
import UserIcon from 'shared-utils/assets/icon-user.svg';
import NFTIcon from 'shared-utils/assets/icon-nft.svg';
import {
  HOME,
  BLOCKS,
  VALIDATORS,
  TRANSACTIONS,
  TOKENS,
  NFTS,
} from '@utils/go_to_page';

export const getMenuItems = () => {
  const iconProps = {
    width: 24,
    height: 24,
  };

  return [
    {
      key: 'overview',
      url: HOME,
      icon: <HomeIcon {...iconProps} />,
    },
    {
      key: 'blocks',
      url: BLOCKS,
      icon: <BlockIcon {...iconProps} />,
    },
    {
      key: 'validators',
      url: VALIDATORS,
      icon: <UserIcon {...iconProps} />,
    },
    {
      key: 'transactions',
      url: TRANSACTIONS,
      icon: <TransactionIcon {...iconProps} />,
    },
    {
      key: 'tokens',
      url: TOKENS,
      icon: <NFTIcon {...iconProps} />,
    },
    {
      key: 'nfts',
      url: NFTS,
      icon: <TokenIcon {...iconProps} />,
    },
  ];
};
