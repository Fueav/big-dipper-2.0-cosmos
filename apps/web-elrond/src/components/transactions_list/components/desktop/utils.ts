export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'hash',
    width: 15,
  },
  {
    key: 'type',
    width: 20,
  },
  {
    key: 'shard',
    width: 15,
  },
  {
    key: 'from',
    width: 15,
  },
  {
    key: 'to',
    width: 15,
  },
  {
    key: 'status',
    width: 10,
  },
  {
    key: 'time',
    width: 10,
    align: 'right',
  },
];
