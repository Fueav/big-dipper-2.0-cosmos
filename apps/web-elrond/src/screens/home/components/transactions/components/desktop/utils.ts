export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'hash',
    width: 20,
  },
  {
    key: 'type',
    width: 20,
  },
  {
    key: 'from',
    width: 15,
  },
  {
    key: 'to',
    width: 15,
  },
  {
    key: 'status',
    width: 15,
  },
  {
    key: 'time',
    width: 15,
    align: 'right',
  },
];
