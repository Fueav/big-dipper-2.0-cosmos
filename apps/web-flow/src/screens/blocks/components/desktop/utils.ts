export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'height',
    width: 25,
  },
  {
    key: 'hash',
    width: 30,
  },
  {
    key: 'txs',
    align: 'right',
    width: 20,
  },
  {
    key: 'time',
    align: 'right',
    width: 25,
  },
];
