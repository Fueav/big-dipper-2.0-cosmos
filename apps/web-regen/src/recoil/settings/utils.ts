import { defaultTheme, isThemeSupported, themeDictionary } from '@styles';
import chainConfig from 'ui/chainConfig';
import { Theme } from './types';

// ================================
// CONSTANTS
// ================================

export const THEME_LIST: Theme[] = chainConfig.themes.themeList;

export const THEME_DICTIONARY = themeDictionary;

export const getThemeTemplate = (theme: Theme) => {
  if (isThemeSupported(theme)) {
    return THEME_DICTIONARY[theme];
  }
  return defaultTheme;
};

export const DATE_LIST = ['locale', 'utc'];

export const TX_LIST = ['compact', 'detailed'];
