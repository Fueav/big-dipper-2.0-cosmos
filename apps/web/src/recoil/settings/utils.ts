import { lightTemplate, themeDictionary } from '@styles';
import chainConfig from 'ui/dist/chainConfig';
import { Theme } from './types';

// ================================
// CONSTANTS
// ================================

export const THEME_LIST: Theme[] = chainConfig.themes.themeList as Theme[];

export const THEME_DICTIONARY = themeDictionary;

export const getThemeTemplate = (theme: Theme) => {
  if (THEME_DICTIONARY[theme]) {
    return THEME_DICTIONARY[theme];
  }
  return lightTemplate;
};

export const DATE_LIST = ['locale', 'utc'];

export const TX_LIST = ['compact', 'detailed'];
