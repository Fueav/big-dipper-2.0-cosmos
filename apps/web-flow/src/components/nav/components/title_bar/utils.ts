import { formatNumber } from 'ui/utils/format_token';
import numeral from 'numeral';

export const formatMarket = (data: { price: number; supply: TokenUnit; marketCap: number }) => {
  const price = data.price !== null ? `$${numeral(data.price).format('0.00')}` : 'N/A';
  const supply =
    data.supply !== null
      ? `${formatNumber(data.supply.value, 2)} ${data.supply.displayDenom.toUpperCase()}`
      : 'N/A';
  const marketCap =
    data.marketCap !== null ? `$${formatNumber(data.marketCap.toString(), 2)}` : 'N/A';

  return [
    {
      key: 'price',
      data: price,
    },
    {
      key: 'supply',
      data: supply,
    },
    {
      key: 'marketCap',
      data: marketCap,
    },
  ];
};
