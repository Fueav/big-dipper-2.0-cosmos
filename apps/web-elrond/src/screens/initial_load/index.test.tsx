import React from 'react';
import renderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil';
import { MockTheme, wait } from '@tests/utils';
import InitialLoad from '.';

// ==================================
// unit tests
// ==================================
describe('screen: InitialLoad', () => {
  it('matches snapshot', async () => {
    let tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = null;

    renderer.act(() => {
      tree = renderer
        .create(
          <RecoilRoot>
            <MockTheme>
              <InitialLoad />
            </MockTheme>
          </RecoilRoot>
        )
        .toJSON();
    });
    await wait();

    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
