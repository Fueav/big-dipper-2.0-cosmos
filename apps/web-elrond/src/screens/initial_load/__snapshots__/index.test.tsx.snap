// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: InitialLoad matches snapshot 1`] = `null`;
